ironfish/src/rpc/routes/wallet/types.ts

```diff
@@ -72,3 +75,41 @@ export async function getAssetBalanceDeltas(
 
   return assetBalanceDeltas
 }
+
+export async function getAccountDecryptedNotes(
+  node: IronfishNode,
+  account: Account,
+  transaction: TransactionValue,
+): Promise<RpcAccountDecryptedNote[]> {
+  const notesByAccount = await node.wallet.decryptNotes(transaction.transaction, null, true, [
+    account,
+  ])
+  const notes = notesByAccount.get(account.id) ?? []
+
+  const serializedNotes: RpcAccountDecryptedNote[] = []
+  for await (const decryptedNote of notes) {
+    const noteHash = decryptedNote.hash
+    const decryptedNoteForOwner = await account.getDecryptedNote(noteHash)
+
+    const isOwner = !!decryptedNoteForOwner
+    const spent = decryptedNoteForOwner ? decryptedNoteForOwner.spent : false
+    const note = decryptedNoteForOwner
+      ? decryptedNoteForOwner.note
+      : new Note(decryptedNote.serializedNote)
+
+    const asset = await node.chain.getAssetById(note.assetId())
+
+    serializedNotes.push({
+      isOwner,
+      owner: note.owner(),
+      memo: note.memo(),
+      value: CurrencyUtils.encode(note.value()),
+      assetId: note.assetId().toString('hex'),
+      assetName: asset?.name.toString('hex') || '',
```

Comment:
In what cases won't the asset be defined? Wouldn't it be a corrupt database if the asset isn't found but the note is decrypted?

Reply:
this logic came before assets were stored in the wallet db and looks the asset up from the chain database. if the note is from a pending mint transaction then the asset might not have been added to the chain yet.

we may be able to update this to use the wallet database instead of the chain now that it's available